### ISSUE TITLE
Product page view crash

### ISSUE THREAD

**Issue by mrrobotisback** [Smile-SA/magento2-module-custom-entity-product-link#57]
Create a product attribute with frontend => CustomEntity::class and input => CustomEntityAttributeInterface::ENTITY_TYPE_CODE

When more than 400 associations are created the product page view crashes.
Increasing the resources is not an option.
The problem resides on this plugin:
smile/module-custom-entity-product-link/Plugin/Block/Product/ViewPlugin.php

**Comment by guvra** [comment]
The following commit reduces the size of the cache tags:
https://github.com/Smile-SA/magento2-module-custom-entity/commit/033843e424667ea38005b715746a5bed209cbc9c

**Comment by mrrobotisback** [comment]
Sorry but this commit only slows down the problem, it is not a solution.

**Comment by guvra** [comment]
> Sorry but this commit only slows down the problem, it is not a solution. The solution would be to use a hash.

The problem is exactly the same as all other entities of Magento (cf. the countless issues on the magento 2 official repo about the same problem for products). The only solution on our side is to use a cache tag that is as short as possible.

Fixing the issue for good is something that can only be done in the Magento core (and would require them to rewrite their whole caching mechanism).

Your commit that injected a hash in the block breaks the caching mechanism, so of course the issue is "fixed", you disabled the cache alltogether by doing this.

**Comment by mrrobotisback** [comment]
Can you show me the exact place where these cache tags are used in Magento core ?

**Comment by guvra** [comment]
Caching in Magento is extremely complex, but there's a good explaination here:
https://inviqa.com/blog/how-full-page-cache-works-magento-commerce

The cache tag variable is hardcoded in the abstract model:
https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Model/AbstractModel.php#L104

**Comment by mrrobotisback** [comment]
Yes but the step I am missing is why with the hash the cache is broken and therefore nothing is cached on the product page anymore?

**Comment by guvra** [comment]
It's the purge directive sent to Varnish that the commit breaks (because the values injected in getIdentities don't match the cache tags of the model).

**Comment by guvra** [comment]
Cache tags names were shortened in https://github.com/Smile-SA/magento2-module-custom-entity/pull/70
